### PR TITLE
chore(sql): add varchar all-not-equal (<>all) function

### PR DESCRIPTION
**Pull request by eugenels**
Note: The testing set `'{abc,xyz}'::text[]` is still parsed as a string array type but converted into UTF-8 upon function initialisation

### CODE REVIEW

File: core/src/test/java/io/questdb/test/griffin/engine/functions/bool/AllNotEqStrFunctionFactoryTest.java [hunk 1]
```diff
@@ -76,6 +76,23 @@ public void testMatch() throws Exception {
...
+    @Test
+    public void testMatchVarcharColumn() throws Exception {
+        assertQuery(
+                "a\n" +
+                        "aaa\n" +
+                        "aaa\n" +
+                        "bbb\n" +
+                        "ccc\n" +
+                        "ccc\n",
+                "select * from tab where a <> all('{abc,xyz}'::text[])",
```
Comment (bluestreak01) [inline core/src/test/java/io/questdb/test/griffin/engine/functions/bool/AllNotEqStrFunctionFactoryTest.java:88]: what is this magic?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 26 / 27 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/AllNotEqVarcharFunctionFactory.java|26|27|96.30%|

File: core/src/test/java/io/questdb/test/griffin/engine/functions/bool/AllNotEqStrFunctionFactoryTest.java (hunk 1 above)
Comment (eugenels) [reply core/src/test/java/io/questdb/test/griffin/engine/functions/bool/AllNotEqStrFunctionFactoryTest.java:88]: another overload for (varchar, string_array). however, I'm not sure if it's useful

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 26 / 27 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/AllNotEqVarcharFunctionFactory.java|26|27|96.30%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 26 / 27 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/AllNotEqVarcharFunctionFactory.java|26|27|96.30%|

**Review by bluestreak01 — CHANGES_REQUESTED** [review]
implementation is tested with ASCII characters only, please include tests containing multi-byte characters

File: core/src/main/java/io/questdb/griffin/engine/functions/bool/AllNotEqVarcharFunctionFactory.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        @Override
+        public Function getArg() {
+            return arg;
+        }
+
+        @Override
+        public boolean getBool(Record rec) {
+            final Utf8Sequence str = arg.getVarcharA(rec);
+            return str != null && set.excludes(str);
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/bool/AllNotEqVarcharFunctionFactory.java:90]: null check seems to be redundant

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/engine/functions/bool/AllNotEqVarcharFunctionFactory.java (hunk 2 above)
Comment (eugenels) [reply core/src/main/java/io/questdb/griffin/engine/functions/bool/AllNotEqVarcharFunctionFactory.java:90]: this additional null check changes the semantics of the function. 
the `hasNull` logic is built into the `CharSequenceHashSet`, but this check is more explicit

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 26 / 27 (96.30%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/engine/functions/bool/AllNotEqVarcharFunctionFactory.java|26|27|96.30%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**